File: server/mailer.js
process.env["NODE_TLS_REJECT_UNAUTHORIZED"] = 0;
const nodemailer = require('nodemailer');
const fs = require("fs");

const config = JSON.parse(fs.readFileSync("./config.json"))

var transporter = nodemailer.createTransport(config.EMAIL.SMTP);

async function send_email_otp(recipient, data) {
    try {
        var mailOptions = {
            from: {
                name: 'GeoMedia app',
                address: 'albmor.dev@gmail.com'
            },
            to: recipient,
            subject: "GeoMedia app - OTP",
            attachments: []
        };
        let htmlBody = fs.readFileSync("./email_body.html", 'utf-8')
        htmlBody = htmlBody?.replaceAll("$USERNAME", data?.USERNAME)
        htmlBody = htmlBody?.replaceAll("$OTP", data?.OTP.toString())

        mailOptions.html = htmlBody;
        let x = await transporter.sendMail(mailOptions);

        return [true, x.response]

    } catch (error) {
        console.log(error)
        return [false, error]
    }
}


send_email_otp("morini99@icloud.com",{"USERNAME":"hey","OTP":"05050"})

module.exports = {
    send_email_otp
}

File: server/mailer.py
import smtplib
import ssl
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText
import json

# Load the SMTP config 
with open('./config.json') as f:
    config = json.load(f)

def send_email_otp(recipient, data):
    try:
        # SMTP settings from config
        smtp_config = config["EMAIL"]["SMTP"]
        host = smtp_config["host"]
        port = smtp_config["port"]
        user = smtp_config["auth"]["user"]
        password = smtp_config["auth"]["pass"]
        
        # Prepare the email
        msg = MIMEMultipart("alternative")
        msg["Subject"] = "GeoMedia app - OTP"
        msg["From"] = f"GeoMedia app <{user}>"
        msg["To"] = recipient
        
        # Load the HTML template
        with open("./email_body.html", "r", encoding="utf-8") as f:
            html_body = f.read()
        
        # Replace placeholders
        html_body = html_body.replace("$USERNAME", str(data.get("USERNAME", "")))
        html_body = html_body.replace("$OTP", str(data.get("OTP", "")))

        # Attach HTML content
        msg.attach(MIMEText(html_body, "html"))

        # Establish connection with the Gmail SMTP server using STARTTLS
        with smtplib.SMTP(host, port) as server:
            # Upgrade connection to secure using STARTTLS
            server.starttls(context=ssl.create_default_context())
            server.login(user, password)  # Login using the provided credentials
            server.sendmail(user, recipient, msg.as_string())  # Send the email

        return [True, "Email sent successfully"]

    except Exception as error:
        print(f"Error: {error}")
        return [False, str(error)]